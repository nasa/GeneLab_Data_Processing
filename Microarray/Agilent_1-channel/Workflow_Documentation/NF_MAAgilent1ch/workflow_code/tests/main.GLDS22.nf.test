nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "core"

    test("GLDS-22: Should run without failures. First Test Case for Arabidopsis") {

        when {
            params {
                // define parameters here. Example: 
                osdAccession = "OSD-22"
                gldsAccession = "GLDS-22"
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                            path("${params.outputDir}/${params.gldsAccession}/Metadata").list(),
                            path("${params.outputDir}/${params.gldsAccession}/00-RawData").list(),
                            path("${params.outputDir}/${params.gldsAccession}/01-limma_NormExp").list(),
                            path("${params.outputDir}/${params.gldsAccession}/02-limma_DGE").list(),
                            path("${params.outputDir}/${params.gldsAccession}/GeneLab/software_versions.md")
                            ).match()
        }

    }

}
