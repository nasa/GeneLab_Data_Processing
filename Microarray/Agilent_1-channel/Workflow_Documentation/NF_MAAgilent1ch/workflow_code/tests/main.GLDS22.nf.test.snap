{
    "GLDS-22: Should run without failures. First Test Case for Arabidopsis": {
        "content": [
            [
                "OSD-22_metadata_OSD-22-ISA.zip:md5,54b9d89f37c99cc3b782ecb6b4f31c66",
                "OSD-22_microarray_v0_runsheet.csv:md5,315cca3328a4fc58c0f2cb2862ec92e8"
            ],
            [
                "raw_intensities.csv:md5,b796344835c59716d224f79d0a47abb6"
            ],
            [
                "normalized_expression.csv:md5,031169ae9804ca0f97e0a5d530a40e88"
            ],
            [
                "SampleTable.csv:md5,e06ec67d73d670aa67efaef842721895",
                "contrasts.csv:md5,8d35c6eb5c8f554f902b26b782af9ecf",
                "differential_expression.csv:md5,2c50db17385bef00f4f2c0a1a4b89606",
                "visualization_PCA_table.csv:md5,588788b7b14c555ec34f1db88130cea1"
            ],
            "software_versions.md:md5,c39595e6b3b78288520518e77223b285"
        ],
        "timestamp": "2023-04-28T20:59:14+0000"
    }
}